sdk/openai/Azure.AI.OpenAI/src/Custom/OpenAIClient.cs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable disable
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.AI.OpenAI.Models;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.AI.OpenAI
+{
+    // Data plane generated client.
+    /// <summary> Azure OpenAI APIs for completions and search. </summary>
+    public partial class OpenAIClient
+    {
+        private readonly string _completionsDeploymentId;
+        private readonly string _embeddingsDeploymentId;
+
+        /// <summary> Initializes a new instance of OpenAIClient. </summary>
+        /// <param name="endpoint">
+        /// Supported Cognitive Services endpoints (protocol and hostname, for example:
+        /// https://westus.api.cognitive.microsoft.com).
+        /// </param>
+        /// <param name="credential"> A credential used to authenticate to an Azure Service. </param>
+        /// <param name="completionsDeploymentId"> default deployment id to use for completions </param>
+        /// <param name="embeddingsDeploymentId"> default deployment id to use for embeddings </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="endpoint"/> or <paramref name="credential"/> is null. </exception>
+        public OpenAIClient(Uri endpoint, TokenCredential credential, string completionsDeploymentId, string embeddingsDeploymentId = null) : this(endpoint, credential, completionsDeploymentId, embeddingsDeploymentId, new OpenAIClientOptions())
```

Comment:
I think we wanted to add these into the options class vs ctor parameters?

@tg-msft ^

Reply:
No, I think we want them on the .ctor because they're required to use the Azure service correctly.  I'd also move them before the `credential` because they're part of the "addressing the resource" set of params.  My bigger concern is adding one per operation though.  OpenAI has a decent number of operations today and this will get messy quickly.

Instead, I'd suggest something like:

```C#
public OpenAIClient(Uri endpoint, string deploymentId, TokenCredential credential)
  : this(endpoint, deploymentId, credential, null)
{
}

public OpenAIClient(Uri endpoint, string deploymentId, TokenCredential credential, OpenAIClientOptions options)
{
    // ...
    options ??= new OpenAIClientOptions();
    // ...
    _completionsDeploymentId = options.CompletionsDeploymentId ?? deploymentId;
    _embeddingsDeploymentId = options.EmbeddingsDeploymentId ?? deploymentId;
}
```

Which lets people pass a single global `deploymentId` for all operations easily but still customize the `deploymentId` globally per operation type via the `OpenAIClientOptions` or per invocation via `*Options` parameters.  The  `OpenAIClientOptions` changes could be skipped for preview if we're low on time because the default .ctor value could still be overridden per invocation.

We can also talk in the future about creating a `OpenAIClientOptions.DefaultDeployments` model type/property if we want to group all the deployments in a single place so we don't have them sprinkled throughout the client options.

Reply:
Updated the constructor to match this comment (no changes to OpenAIClientOptions until after preview).

Reply:
@tg-msft I assume in there we wanted the ctor to be mirroed for both the TokenCredential set and the AzureKeyCredential set?  Also did we want to remove the `deploymentId` from the method signatures?  If so should we remove the ctor that doesn't have the `deploymentId` argument?

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/33789

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/33815